File: scripts/layout_guide.py
#!env python3

from generate import Component, Property, Method, Return, Param, init_method


if __name__ == "__main__":
    w = Component(
        Type="appkit.LayoutGuide",
        super_type='foundation.Object',
        description="a rectangular area that can interact with Auto Layout",
        properties=[
            Property(name='frame', Type='foundation.Rect', readonly=True,
                     description='the layout guide’s frame in its owning view’s coordinate system'),

            Property(name='bottomAnchor', Type='appkit.LayoutYAxisAnchor', readonly=True,
                     description='a layout anchor representing the bottom edge of the view’s frame'),
            Property(name='centerXAnchor', Type='appkit.LayoutXAxisAnchor',readonly=True,
                     description='a layout anchor representing the horizontal center of the view’s frame'),
            Property(name='centerYAnchor', Type='appkit.LayoutYAxisAnchor',readonly=True,
                     description='a layout anchor representing the vertical center of the view’s frame'),
            Property(name='heightAnchor', Type='appkit.LayoutDimension', readonly=True, description=''),
            Property(name='leadingAnchor', Type='appkit.LayoutXAxisAnchor', readonly=True, description=''),
            Property(name='leftAnchor', Type='appkit.LayoutXAxisAnchor', readonly=True, description=''),
            Property(name='rightAnchor', Type='appkit.LayoutXAxisAnchor', readonly=True, description=''),
            Property(name='topAnchor', Type='appkit.LayoutYAxisAnchor', readonly=True, description=''),
            Property(name='trailingAnchor', Type='appkit.LayoutXAxisAnchor', readonly=True, description=''),
            Property(name='widthAnchor', Type='appkit.LayoutDimension', readonly=True, description=''),
            Property(name='owningView', Type='appkit.View', readonly=True, description=''),
        ],
        methods=[
            init_method(name="init", Type="appkit.LayoutGuide"),
        ]
    )
    w.generate_code()

File: appkit/layout_guide.m
#import <AppKit/AppKit.h>
#import "layout_guide.h"

NSRect LayoutGuide_Frame(void* ptr) {
	NSLayoutGuide* layoutGuide = (NSLayoutGuide*)ptr;
	return [layoutGuide frame];
}

void* LayoutGuide_BottomAnchor(void* ptr) {
	NSLayoutGuide* layoutGuide = (NSLayoutGuide*)ptr;
	return [layoutGuide bottomAnchor];
}

void* LayoutGuide_CenterXAnchor(void* ptr) {
	NSLayoutGuide* layoutGuide = (NSLayoutGuide*)ptr;
	return [layoutGuide centerXAnchor];
}

void* LayoutGuide_CenterYAnchor(void* ptr) {
	NSLayoutGuide* layoutGuide = (NSLayoutGuide*)ptr;
	return [layoutGuide centerYAnchor];
}

void* LayoutGuide_HeightAnchor(void* ptr) {
	NSLayoutGuide* layoutGuide = (NSLayoutGuide*)ptr;
	return [layoutGuide heightAnchor];
}

void* LayoutGuide_LeadingAnchor(void* ptr) {
	NSLayoutGuide* layoutGuide = (NSLayoutGuide*)ptr;
	return [layoutGuide leadingAnchor];
}

void* LayoutGuide_LeftAnchor(void* ptr) {
	NSLayoutGuide* layoutGuide = (NSLayoutGuide*)ptr;
	return [layoutGuide leftAnchor];
}

void* LayoutGuide_RightAnchor(void* ptr) {
	NSLayoutGuide* layoutGuide = (NSLayoutGuide*)ptr;
	return [layoutGuide rightAnchor];
}

void* LayoutGuide_TopAnchor(void* ptr) {
	NSLayoutGuide* layoutGuide = (NSLayoutGuide*)ptr;
	return [layoutGuide topAnchor];
}

void* LayoutGuide_TrailingAnchor(void* ptr) {
	NSLayoutGuide* layoutGuide = (NSLayoutGuide*)ptr;
	return [layoutGuide trailingAnchor];
}

void* LayoutGuide_WidthAnchor(void* ptr) {
	NSLayoutGuide* layoutGuide = (NSLayoutGuide*)ptr;
	return [layoutGuide widthAnchor];
}

void* LayoutGuide_OwningView(void* ptr) {
	NSLayoutGuide* layoutGuide = (NSLayoutGuide*)ptr;
	return [layoutGuide owningView];
}

void* LayoutGuide_NewLayoutGuide() {
	NSLayoutGuide* layoutGuide = [NSLayoutGuide alloc];
	return [[layoutGuide init] autorelease];
}
